core/EE_System.core.php

```diff
@@ -1161,15 +1161,9 @@ public function core_loaded_and_ready()
         // integrate WP_Query with the EE models
         $this->loader->getShared('EE_CPT_Strategy');
         do_action('AHEE__EE_System__core_loaded_and_ready');
-        // load_espresso_template_tags
-        if (($this->request->isFrontend()
-             || $this->request->isIframe()
-             || $this->request->isFeed()
-             ||  $this->request->isAjax()
-            ) && is_readable(EE_PUBLIC . 'template_tags.php')
-        ) {
-            require_once EE_PUBLIC . 'template_tags.php';
-        }
+        // always load template tags, because it's faster than checking if it's a front-end request, and many page
+        // builders require these even on the front-end
+        require_once EE_PUBLIC . 'template_tags.php';
```

Comment:
should there still be a `is-readable` check though?

Reply:
good question. Under what circumstances, besides someone hacked EE core by removing or moving that file, would the file NOT be readable?
I see the constant `EE_PUBLIC` is defined in `core/espresso_definitions.php` and is always `define('EE_PUBLIC', EE_PLUGIN_DIR_PATH . 'public' . DS);`, where `EE_PLUGIN_DIR_PATH` is always the plugin's directory path. (Maybe at one point it was dynamic, but from the code I read it doesn't get much more static than this.)
From what I read, that file will always be present unless someone has hacked EE core. Right? If so, I'm inclined to not protect against someone hacking EE core, and save a bit of processing to check if the file exists.

Reply:
works for me.